src/sentry/seer/anomaly_detection/store_data.py

```diff
@@ -0,0 +1,187 @@
+        return base_error_response
+
+    snuba_query = SnubaQuery.objects.get(id=rule.snuba_query_id)
+    time_period = int(snuba_query.time_window / 60)
+    historical_data = fetch_historical_data(rule, snuba_query)
+
+    if not historical_data:
+        base_error_response.reason = "No historical data available. Cannot create rule."
+        return base_error_response
+
+    formatted_data = format_historical_data(historical_data)
+
+    if not rule.sensitivity or not rule.seasonality or rule.threshold_type is None:
+        # this won't happen because we've already gone through the serializer, but mypy insists
+        base_error_response.reason = (
+            "Cannot create rule - missing expected configuration for a dynamic alert."
+        )
+        return base_error_response
+
+    ad_config = ADConfig(
+        time_period=time_period,
+        sensitivity=rule.sensitivity,
+        direction=translate_direction(rule.threshold_type),
+        expected_seasonality=rule.seasonality,
+    )
+    alert = Alert(id=rule.id)
+    body = StoreDataRequest(
+        organization_id=rule.organization.id,
+        project_id=project_id,
+        alert=alert,
+        config=ad_config,
+        timeseries=formatted_data,
+    )
+    try:
+        resp = make_signed_seer_api_request(
+            connection_pool=seer_anomaly_detection_connection_pool,
+            path=SEER_ANOMALY_DETECTION_STORE_DATA_URL,
+            body=json.dumps(body).encode("utf-8"),
+        )
+    # See SEER_ANOMALY_DETECTION_TIMEOUT in sentry.conf.server.py
+    except (TimeoutError, MaxRetryError):
+        timeout_text = "Timeout error when hitting Seer store data endpoint"
+        logger.warning(
+            timeout_text,
+            extra={
+                "rule_id": rule.id,
+                "project_id": project_id,
+            },
+        )
+        base_error_response.reason = timeout_text
+        base_error_response.status = status.HTTP_408_REQUEST_TIMEOUT
+        return base_error_response
+
+    # TODO warn if there isn't at least 7 days of data
+    return resp
+
+
+def fetch_historical_data(rule: AlertRule, snuba_query: SnubaQuery) -> SnubaTSResult | None:
+    """
+    Fetch 28 days of historical data from Snuba to pass to Seer to build the anomaly detection model
+    """
+    # TODO: if we can pass the existing timeseries data we have on the front end along here, we can shorten
+    # the time period we query and combine the data
+    NUM_DAYS = 28
+    end = timezone.now()
+    start = end - timedelta(days=NUM_DAYS)
+    granularity = snuba_query.time_window
+
+    dataset_label = snuba_query.dataset
+    if dataset_label == "events":
+        # DATSET_OPTIONS expects the name 'errors'
+        dataset_label = "errors"
+    dataset = get_dataset(dataset_label)
+    project_id = get_project_id_from_rule(rule)
+    if not project_id or not dataset:
+        return None
+
+    historical_data = dataset.timeseries_query(
```

Comment:
This query should match teh subscription query right?

Reply:
yeah it's the same metric and dataset, but we're getting it for 28 days